### PR TITLE
SONARJAVA-3651 S4925: add support for DB2 JDBC Type 4 driver

### PR DESCRIPTION
**Pull request by quentin-jaquier-sonarsource**

### CODE REVIEW

File: java-checks-test-sources/src/main/java/checks/JdbcDriverExplicitLoadingCheck_no_version.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
-class A {
-  void doStuff() {
+package checks;
+
+class JdbcDriverExplicitLoadingCheck_no_version {
+  void doStuff() throws ClassNotFoundException {
     Class.forName("org.h2.Driver"); // Noncompliant [[sc=11;ec=18]] {{Remove this "Class.forName()", it is useless. (sonar.java.source not set. Assuming 6 or greater.)}}
     Class.forName("java.lang.String");
     Class.forName(javax.print.ServiceUIFactory.DIALOG_UI);
```
Comment (dorian-burihabwa-sonarsource) [inline java-checks-test-sources/src/main/java/checks/JdbcDriverExplicitLoadingCheck_no_version.java:7]: For completeness, we should add a call with a non compliant string defined in the scope of this class

Comment (quentin-jaquier-sonarsource) [reply java-checks-test-sources/src/main/java/checks/JdbcDriverExplicitLoadingCheck_no_version.java:7]: If I understand correctly, you mean to add code like this:

```java
  private static final String DRIVER = "com.mysql.jdbc.Driver";

  void doStuff() throws ClassNotFoundException {
    Class.forName(DRIVER); // Noncompliant
  }
```

In this file? My feeling is that we already test this behavior in the other file, this one is here to test the message with an unknown version and should not duplicate what is tested in the other one.

Does it make sense?

Comment (dorian-burihabwa-sonarsource) [reply java-checks-test-sources/src/main/java/checks/JdbcDriverExplicitLoadingCheck_no_version.java:7]: Then I am wondering why we are using two test-source files with overlapping cases, including this one that has a limited number of test cases.
Couldn't we use a single test-source file for this rule?

Comment (quentin-jaquier-sonarsource) [reply java-checks-test-sources/src/main/java/checks/JdbcDriverExplicitLoadingCheck_no_version.java:7]: In fact, we are testing that the message is not the same when the version of java is not set. But I agree that if we really want to keep this file minimal, we should remove the line `Class.forName(javax.print.ServiceUIFactory.DIALOG_UI);`.

Comment (dorian-burihabwa-sonarsource) [reply java-checks-test-sources/src/main/java/checks/JdbcDriverExplicitLoadingCheck_no_version.java]: Good plan

**Pull request merged**